dev/bots/utils.dart

```diff
@@ -71,6 +71,13 @@ final Map<String,String> localEngineEnv = <String, String>{};
 /// configuration) -- prefilled with the arguments passed to test.dart.
 final List<String> flutterTestArgs = <String>[];
 
+/// Whether execution should be simulated for debugging purposes.
+///
+/// When `true`, calls to [runCommand] print to [io.stdout] instead of running
+/// the process. This is useful for determing what an invocation of `test.dart`
+/// _might_ due if not invoked with `--dry-mode`, or otherwise determine what
+/// the different test shards and sub-shards are configured as.
+bool dryRun = false;
```

Comment:
Fair question. The answer is no, as there are no unit tests where this is true (there is an integration test where this is set to true implicitly with `--dry-run`) but it's an isolated process.

I modified the API a bit to make it more clear how this will be used.